src/mmw/js/src/analyze/models.js

```diff
@@ -141,89 +141,210 @@ var AnalyzeTaskCollection = Backbone.Collection.extend({
+                    area_of_interest: aoi,
+                    wkaoi: wkaoi,
+                    taskName: "analyze/terrain"
+                }
+            ]
         },
         {
             name: "climate",
             displayName: "Climate",
-            area_of_interest: aoi,
-            wkaoi: wkaoi,
-            taskName: "analyze/climate",
-            enabledForCatalogMode: true,
+            tasks: [
+                {
+                    name: "climate",
+                    area_of_interest: aoi,
+                    wkaoi: wkaoi,
+                    taskName: "analyze/climate",
+                    enabledForCatalogMode: true,
+                },
+            ]
         },
         {
             name: "pointsource",
             displayName: "Pt Sources",
-            area_of_interest: aoi,
-            wkaoi: wkaoi,
-            taskName: "analyze/pointsource"
+            tasks: [
+                {
+                    name: "pointsource",
+                    area_of_interest: aoi,
+                    wkaoi: wkaoi,
+                    taskName: "analyze/pointsource"
+                }
+            ]
         },
         {
             name: "animals",
             displayName: "Animals",
-            area_of_interest: aoi,
-            wkaoi: wkaoi,
-            taskName: "analyze/animals"
+            tasks: [
+                {
+                    name: "animals",
+                    area_of_interest: aoi,
+                    wkaoi: wkaoi,
+                    taskName: "analyze/animals"
+                }
+            ]
         },
         {
             name: "catchment_water_quality",
             displayName: "Water Qual",
-            area_of_interest: aoi,
-            wkaoi: wkaoi,
-            taskName: "analyze/catchment-water-quality"
-        },
-        {
-            name: "protected_lands",
-            displayName: "Protected Lands",
-            area_of_interest: aoi,
-            wkaoi: wkaoi,
-            taskName: "analyze/protected-lands"
+            tasks: [
+                {
+                    name: "catchment_water_quality",
+                    area_of_interest: aoi,
+                    wkaoi: wkaoi,
+                    taskName: "analyze/catchment-water-quality"
+                }
+            ]
         },
     ];
 
     // Remove analyses which aren't available for catalog search mode
     if (settings.get('data_catalog_enabled')) {
-        tasks = _.filter(tasks, { enabledForCatalogMode: true });
+        var filteredTaskGroups = [];
+        _.foreach(taskGroups, function(tg) {
+            var filteredTasks = _.filter(tg.tasks, { enabledForCatalogMode: true });
+            if(filteredTasks.length > 0) {
+                tg.tasks = filteredTasks;
+                filteredTaskGroups.push(tg);
+            }
+        });
+
+        taskGroups = filteredTaskGroups;
     }
 
-    return new AnalyzeTaskCollection(tasks);
+    taskGroups = _.map(taskGroups, function (tg) {
```

Comment:
Prefer eliding this in favor of the conversion in the `AnalyzeTaskGroupModel` constructor:

```diff
diff --git a/src/mmw/js/src/analyze/models.js b/src/mmw/js/src/analyze/models.js
index a1b7252c..51492b5a 100644
--- a/src/mmw/js/src/analyze/models.js
+++ b/src/mmw/js/src/analyze/models.js
@@ -153,9 +153,8 @@ var AnalyzeTaskGroupModel = Backbone.Model.extend({
     initialize: function() {
         var tasks = this.get('tasks');
 
-        if (tasks === null || typeof tasks === 'string') {
-            this.set('tasks', new AnalyzeTaskCollection());
-        }
+        // Convert tasks array to collection
+        this.set('tasks', new AnalyzeTaskCollection(tasks));
     },
 
     /**
@@ -332,11 +331,6 @@ function createAnalyzeTaskGroupCollection(aoi, wkaoi) {
             .value();
     }
 
-    taskGroups = _.map(taskGroups, function (tg) {
-        tg.tasks = new AnalyzeTaskCollection(tg.tasks);
-        return tg;
-    });
-
     return new AnalyzeTaskGroupCollection(taskGroups);
 }
 
```

Reply:
I added this in a846042 which was needed to get the tests to run correctly.